internal/connector/internal/config/connectors.go

```diff
@@ -51,20 +58,82 @@ func NewConnectorsConfig() *ConnectorsConfig {
 
 func (c *ConnectorsConfig) AddFlags(fs *pflag.FlagSet) {
 	fs.StringArrayVar(&c.ConnectorCatalogDirs, "connector-catalog", c.ConnectorCatalogDirs, "Directory containing connector catalog entries")
+	fs.StringArrayVar(&c.ConnectorMetadataDirs, "connector-metadata", c.ConnectorMetadataDirs, "Directory containing connector metadata configuration files")
 	fs.DurationVar(&c.ConnectorEvalDuration, "connector-eval-duration", c.ConnectorEvalDuration, "Connector eval duration in golang duration format")
 	fs.StringArrayVar(&c.ConnectorEvalOrganizations, "connector-eval-organizations", c.ConnectorEvalOrganizations, "Connector eval organization IDs")
 	fs.BoolVar(&c.ConnectorNamespaceLifecycleAPI, "connector-namespace-lifecycle-api", c.ConnectorNamespaceLifecycleAPI, "Enable APIs to create, update, delete non-eval Namespaces")
 	fs.BoolVar(&c.ConnectorEnableUnassignedConnectors, "connector-enable-unassigned-connectors", c.ConnectorEnableUnassignedConnectors, "Enable support for 'unassigned' state for Connectors")
 }
 
 func (c *ConnectorsConfig) ReadFiles() error {
-	typesLoaded := map[string]string{}
-	var values []ConnectorCatalogEntry
+	// read metadata first to merge with catalog next
+	connectorMetadata, err := c.readConnectorMetadata()
+	if err != nil {
+		return err
+	}
+
+	c.CatalogEntries, err = c.readConnectorCatalog(connectorMetadata)
+	if err != nil {
+		return err
+	}
+
+	remainingIds := len(connectorMetadata)
```

Comment:
The fact that `connectorMetadata` gets modified inside `readConnectorCatalog` doesn't help for clarity, see other comment a but further.

Reply:
See previous comment.